### PR TITLE
diff: add unit test && make importer a pkg

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
1. add unit test for  diff
2. make importer a pkg

### What is changed and how it works?
add unit test:
1. test table struct
2. test table data when data is equal
3. test table with different data
4. test fix sql

Tests 
 - Unit test

### CODE REVIEW

File: pkg/diff/diff_test.go [hunk 1]
```diff
@@ -78,3 +85,185 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
...
+	}
+
+	tableDiff := createTableDiff(dbConn)
+	structEqual, dataEqual, err := tableDiff.Equal(context.Background(), writeSqls)
+	c.Assert(err, IsNil)
+	c.Assert(structEqual, Equals, true)
+	c.Assert(dataEqual, Equals, true)
+
+	// update data and then compare data
+	_, err = dbConn.Exec("update test.testb set d = \"abc\" where a = \"2045-12-29\"")
```
Comment (kennytm) [inline pkg/diff/diff_test.go]:
```suggestion
	_, err = dbConn.Exec("update test.testb set d = 'abc' where a = '2045-12-29'")
```

No need to escape :)

File: pkg/diff/diff_test.go [hunk 2]
```diff
@@ -60,21 +67,203 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
...
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int, `name2` varchar(24))",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int)",
```
Comment (IANTHEREAL) [inline pkg/diff/diff_test.go:121]: change this case to check column type

File: pkg/diff/diff_test.go [hunk 3]
```diff
@@ -60,21 +67,203 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
...
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
```
Comment (IANTHEREAL) [inline pkg/diff/diff_test.go:125]: lack of a case: check index

**Comment by IANTHEREAL** [comment]
Rest LGTM. we can use go test coverage tool to get the logic code branchs we don't test

File: pkg/diff/diff_test.go [hunk 4]
```diff
@@ -60,21 +67,203 @@ func (*testDiffSuite) TestGenerateSQLs(c *C) {
...
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int, `name2` varchar(24))",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
+			false,
+		}, {
+			"CREATE TABLE `test`.`testa`(`id` int, `name` varchar(24))",
+			"CREATE TABLE `test`.`testb`(`id` int)",
+			"DROP TABLE `test`.`testa`",
+			"DROP TABLE `test`.`testb`",
```
Comment (zier-one) [inline pkg/diff/diff_test.go:123]: maybe we should test `create table test.texta (column_1 int default 123 null comment '321');` and  `create table test.textb (column_1 int);`

Comment (zier-one) [reply pkg/diff/diff_test.go:123]: and other ColumnOption

File: pkg/diff/test.sql [hunk 5]
```diff
@@ -0,0 +1,10001 @@
+INSERT INTO `test`.`testa` VALUES
```
Comment (july2993) [inline pkg/diff/test.sql]: how this file come,  why not generate the test data in your test

Comment (WangXiangUSTC) [reply pkg/diff/test.sql]: done, use importer to generate data

File: pkg/diff/diff_test.go (hunk 2 above)
Comment (WangXiangUSTC) [reply pkg/diff/diff_test.go:121]: add a new case

File: pkg/diff/diff_test.go (hunk 4 above)
Comment (WangXiangUSTC) [reply pkg/diff/diff_test.go:123]: don't support check column option now

File: pkg/diff/diff_test.go (hunk 3 above)
Comment (WangXiangUSTC) [reply pkg/diff/diff_test.go:125]: the first case had unique key

**Comment by zier-one** [comment]
LGTM

**Comment by IANTHEREAL** [comment]
LGTM

**Pull request merged**